### PR TITLE
Refactored InterationMapping to be MixedRealityInputMapping

### PR DESCRIPTION
**Pull request by StephenHodgson**
First of several PRs addressing changes.

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Definitions/MixedRealityConfigurationProfile.cs [hunk 1]
```diff
@@ -220,29 +220,5 @@ void ISerializationCallbackReceiver.OnAfterDeserialize()
         }
 
         #endregion  ISerializationCallbackReceiver Implementation
-
-        #region Mixed Reality Controller Mapping helpers
-
-        // TODO - needs validation to ensure duplicates are not added?
-
-        public MixedRealityControllerMapping GetControllerMapping(Type controllerType, Handedness hand)
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Definitions/MixedRealityConfigurationProfile.cs]: This is addressed in the next PR.

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Definitions/MixedRealityConfigurationProfile.cs]: Will be moved to base controller in devices refactor. @SimonDarksideJ do you want me to revert until then?

File: Assets/MixedRealityToolkit/_Core/Extensions/CollectionsExtensions.cs [hunk 2]
```diff
@@ -150,13 +150,30 @@ public static T[] ExportDictionaryValuesAsArray<T>(this Dictionary<uint,T> input
...
             return changed;
         }
+
+
+        /// <summary>
+        /// Overload extension to enable getting of an InteractionDefinition of a specific type
+        /// </summary>
+        /// <param name="input">The InteractionDefinition array reference</param>
+        /// <param name="key">The specific DeviceInputType value to query</param>
+        public static MixedRealityInteractionMapping GetInteractionbyType(this MixedRealityInteractionMapping[] input, DeviceInputType key)
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/_Core/Extensions/CollectionsExtensions.cs]: GetInteractionbyType -> GetInteractionByType?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Extensions/CollectionsExtensions.cs]: Fixed in a future PR I caught that too.

**Pull request merged**